src/js/utils/animations.js

```diff
@@ -0,0 +1,139 @@
+import { css, keyframes } from 'styled-components';
+
+const PULSE_SIZES = {
+  xsmall: 1.001,
+  small: 1.01,
+  medium: 1.1,
+  large: 1.5,
+  xlarge: 2,
+};
+
+const SLIDE_SIZES = {
+  xsmall: 1,
+  small: 5,
+  medium: 10,
+  large: 50,
+  xlarge: 200,
+};
+
+const JIGGLE_SIZES = {
+  xsmall: 0.1,
+  small: 1,
+  medium: 5,
+  large: 400,
+  xlarge: 1000,
+};
+
+const ZOOM_SIZES = {
+  xsmall: 0.001,
+  small: 0.01,
+  medium: 0.05,
+  large: 0.1,
+  xlarge: 0.5,
+};
+
+export const animationBounds = (type, size = 'medium') => {
+  if (type === 'draw') {
+    return ['', `stroke-dashoffset: 0`];
+  }
+  if (type === 'fadeIn') {
+    return ['opacity: 0;', 'opacity: 1;'];
+  }
+  if (type === 'fadeOut') {
+    return ['opacity: 1;', 'opacity: 0;'];
+  }
+  if (type === 'jiggle') {
+    const deg = JIGGLE_SIZES[size];
+    return [`transform: rotate(-${deg}deg);`, `transform: rotate(${deg}deg);`];
+  }
+  if (type === 'pulse') {
+    return ['transform: scale(1);', `transform: scale(${PULSE_SIZES[size]})`];
+  }
+  if (type === 'rotateRight') {
```

Comment:
What happens if I give my diagram an animation type of rotateRight? I understand it's not documented but since the component relies on this code will any errors result? does the diagram rotate?

Reply:
Now Fixed: issue where if you gave Diagram any of the animations available the Diagram would carry it out even though it wasn't intended for the component. Handled it now in StyledDiagram.js. Just returns `Not available for Diagram` to be more explicit.